packages/contracts/contracts/bridge/xERC20Hyperlane.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.22;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IXERC20 } from "./interface/IXERC20.sol";
+import { IMailbox } from "./hyperlane/interfaces/IMailbox.sol";
+
+library TypeCasts {
+  // alignment preserving cast
+  function addressToBytes32(address _addr) internal pure returns (bytes32) {
+    return bytes32(uint256(uint160(_addr)));
+  }
+
+  // alignment preserving cast
+  function bytes32ToAddress(bytes32 _buf) internal pure returns (address) {
+    return address(uint160(uint256(_buf)));
+  }
+}
+
+contract xERC20Hyperlane is Ownable {
+  using TypeCasts for address;
+  using TypeCasts for bytes32;
+
+  uint256 public feeBps;
+  mapping(address => mapping(uint32 => address)) public mappedTokens;
```

Comment:
since you're using solidity >=0.8.22, consider using named mapping which will make this nested mapping self-explanatory